cli/bpmetadata/tfconfig.go

```diff
@@ -188,6 +189,43 @@ func parseBlueprintVersion(versionsFile *hcl.File, diags hcl.Diagnostics) (strin
 	return "", nil
 }
 
+// parseBlueprintProviderVersions gets the blueprint provider_versions from the provided config
+// from the provider_meta block
+func parseBlueprintProviderVersions(versionsFile *hcl.File) ([]*ProviderVersion, error) {
+	var v []*ProviderVersion
+	var diags hcl.Diagnostics
+
+	//parse out the required providers from the config
+	var hclModule tfconfig.Module
+	hclModule.RequiredProviders = make(map[string]*tfconfig.ProviderRequirement)
+	hclModuleDiag := tfconfig.LoadModuleFromFile(versionsFile, &hclModule)
+	diags = append(diags, hclModuleDiag...)
+	err := hasHclErrors(diags)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, providerData := range hclModule.RequiredProviders {
+		providerVersion := &ProviderVersion{}
+
+		source := providerData.Source
```

Comment:
nit: is there a reason to assign to source and then validate. IMO flow could be simplified to

```
if providerData.Source == ""{
   ...
}
if len(providerData. VersionConstraints) == 0{
   ...
}
v = append(v, &ProviderVersion{
source: providerData.Source,
version: strings.Join(providerData.VersionConstraints, ", "),
})
```